src/utils.ts

```diff
@@ -41,3 +41,18 @@ export function isFunction(functionToCheck: any): boolean {
     functionToCheck && {}.toString.call(functionToCheck) === "[object Function]"
   );
 }
+
+export async function asyncFind<T>(
+  arr: T[],
+  predicate: (element: T) => Promise<boolean>
+): Promise<T | null> {
+  return arr.reduce(async (result: Promise<T | null>, element: T) => {
+    if (await result) {
+      return result;
+    }
+    if (await predicate(element)) {
+      return element;
+    }
+    return Promise.resolve(null);
+  }, Promise.resolve(null));
+}
```

Comment:
I can't understand what this function does, and whether or not it does it as intended. Can you refactor it to not use `reduce`?